### ISSUE TITLE
[QUESTION] express-session vs cookie-session

### ISSUE THREAD

**Issue by CameronJ** [oskosk/express-socket.io-session#62]
I am just starting to look at using this module, and the example discusses express-session.  In many cases, I often use cookie-session instead of express-session when the use-case doesn't require such a heavy module.

I began working with this module using cookie-session, since that what I had implemented already and everything appears to be working properly.  Is there anything that I should be aware of, which may not be compatible?  Or potentially things that I should verify?

**Comment by CameronJ** [comment]
After some more usage, it appears like the "save" functionality, using autoSave or not, appears like it doesn't work.  Perhaps this is a problem with using "cookie-session".  Calling socket.handshake.session.save() throws this error (paths truncated to relative):
TypeError: res.getHeader is not a function
    at Cookies.set (./node_modules/cookies/index.js:86:21)
    at Session.save (./node_modules/cookie-session/index.js:226:22)
    at Socket.AuthenticationSocket.socket.on (./socket/AuthenticationSockets.js:24:29)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at ./node_modules/socket.io/lib/socket.js:528:12
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)

**Comment by CameronJ** [comment]
Closing this issue, as I had a facepalm moment with this.  I continued to think about the problem and of course it doesn't work... the WebSocket doesn't exactly return an HTTP response which could include a Header for the cookie (reading the cookie could work, writing the cookie would not). I will be switching over to express-session for this implementation.  As the server side session data will allow this manipulation, without altering the cookie.
